ssm.tf

```diff
@@ -9,3 +9,23 @@ resource "aws_ssm_parameter" "this" {
     ignore_changes = [name]
   }
 }
+
+module "ssm_connection" {
+  count                     = var.ssm_create_connection_parameter ? 1 : 0
+  source                    = "github.com/champ-oss/terraform-aws-ssm.git?ref=v1.0.3-f38d8fb"
+  git                       = var.git
+  name                      = var.ssm_prefix_connection
+  enable_kms                = length(var.ssm_shared_principal_arns) > 0 ? true : false
+  enable_ram_permission     = length(var.ssm_shared_principal_arns) > 0 ? true : false
```

Comment:
defaulting var.ssm_shared_principal_arns to null will make this a much cleaner evaluation:

var.ssm_shared_principal_arns != null ? true : false